### DISCUSSION TITLE
Reconfiguring/Fixing Monitor Interface

### DISCUSSION THREAD

**Discussion by sjdwhiting** [Security-Onion-Solutions/securityonion#2849] [Unsupported Versions] [answered]
I was able to get my initial install up and running on virtual box using a bridged adapter for my management interface and I attempted to use a host only adapter for my monitor. I am able to login to the web console from my host machine. I also have two Ubuntu VMs also using the host only adapter so they should be on the same virtual network as the monitor interface (which is set to promiscuous). 

Now that SO is installed, where would I start looking to get more information on my monitor interface and possibly reconfigure it? In the previous versions of SO there was some pretty straightforward utilities for this but I'm not sure where to look in this version.

**Comment by weslambert** [comment]
Monitor interfaces are added to a bond (`bond0`).  We have a tool coming in the next release to make it easier to add an interface to the bond configuration, but before going ahead what is it that you are looking to reconfigure?

**Reply by sjdwhiting** [reply]
Well I was having some issue with the initial network adapter I setup during the install. I ended up going into Virtual Box, adding another adapter, and using nmcli to attach it to `bond0' as show in another post on here. That adapter seemed to work and I started getting traffic showing up on the web console. Not sure what went wrong with the one during setup as they were both host only adapters set to promiscuous.

**Comment by xme** [comment, marked as answer]
I just landed here thanks to Google :-) I'm looking into adding an extra interface to an existing SO setup. I found that a bond interface is created but can I just add the new one to this bond? Any extra step to perform?

**Reply by weslambert** [reply]
Aside from just adding the interface to the bond, you'll want to make sure some other settings are configured.  You can use the following script to add an interface to the bond (ex. so-monitor-add $interface).  It will be available in the next release:

```
#!/bin/bash
. /usr/sbin/so-common
add_interface_bond0() {
	local BNIC=$1
	if [[ -z $MTU ]]; then
		local MTU
		MTU=$(lookup_pillar "mtu" "sensor")
	fi
	local nic_error=0
	# Check if specific offload features are able to be disabled
	for string in "generic-segmentation-offload" "generic-receive-offload" "tcp-segmentation-offload"; do
		if ethtool -k "$BNIC" | grep $string | grep -q "on [fixed]"; then
			echo "The hardware or driver for interface ${BNIC} is not supported, packet capture may not work as expected."
			((nic_error++))
			break
		fi
	done
	case "$2" in
		-v|--verbose)
			local verbose=true
		;;
	esac
	for i in rx tx sg tso ufo gso gro lro; do
		if [[ $verbose == true ]]; then
			ethtool -K "$BNIC" $i off
		else
			ethtool -K "$BNIC" $i off &>/dev/null
		fi
	done
	# Check if the bond slave connection has already been created
	nmcli -f name,uuid -p con | grep -q "bond0-slave-$BNIC"
	local found_int=$?
	if [[ $found_int != 0 ]]; then
		# Create the slave interface and assign it to the bond
		nmcli con add type ethernet ifname "$BNIC" con-name "bond0-slave-$BNIC" master bond0 -- \
			ethernet.mtu "$MTU" \
			connection.autoconnect "yes"
	else
		local int_uuid
		int_uuid=$(nmcli -f name,uuid -p con | sed -n "s/bond0-slave-$BNIC //p" | tr -d ' ')
		nmcli con mod "$int_uuid" \
			ethernet.mtu "$MTU" \
			connection.autoconnect "yes"
	fi
	ip link set dev "$BNIC" arp off multicast off allmulticast off promisc on
	# Bring the slave interface up
	if [[ $verbose == true ]]; then
		nmcli con up "bond0-slave-$BNIC"
	else
		nmcli con up "bond0-slave-$BNIC" &>/dev/null
	fi
	if [ "$nic_error" != 0 ]; then
		return "$nic_error"
	fi
}
add_interface_bond0 "$1"

```
